File: Basics_with_JavaScript/!Exam Preparation/4/04. Balls_Условие_2020.07.18-19.js
function vladi(array){
	let ballQty = Number(array[0]);
    points = 0;
    let red = 0; let orange = 0; let yellow = 0; let white = 0; let black = 0; let out = 0;

    for (let i = 1; i <= ballQty; i++) {
        let color = array[i];
        
        switch (color) {
            case "red": points += 5; red += 1;  break;
            case "orange": points += 10; orange += 1;  break;
            case "yellow": points += 15; yellow += 1;  break;
            case "white": points += 20; white += 1;  break;
            case "black": points /= 2; points = Math.floor(points); black += 1;  break;
            default:
                out +=1;
                continue;
                break;

        }
    }

    console.log(`Total points: ${points}`)
    console.log(`Red balls: ${red}`)
    console.log(`Orange balls: ${orange}`)
    console.log(`Yellow balls: ${yellow}`)
    console.log(`White balls: ${white}`)
    console.log(`Other colors picked: ${out}`)
    console.log(`Divides from black balls: ${black}`)


    }
   
    vladi(["3",
    "white",
    "black",
    "pink"])
    ;
    vladi(["5",
    "red",
    "red",
    "ddd",
    "ddd",
    "ddd"])
    ;

File: Basics_with_Python/Python_Basic_Exam_re/4/04._Balls_Условие_2020.07.18_19.py
import math

number = int(input())
points = 0
red_cnt = 0
orange_cnt = 0
yellow_cnt = 0
white_cnt = 0
others_cnt = 0
black_cnt = 0

for i in range(number):
    ball = input()

    if ball == "red":
        red_cnt += 1
        points += 5
    elif ball == "orange":
        orange_cnt += 1
        points += 10
    elif ball == "yellow":
        yellow_cnt += 1
        points += 15
    elif ball == "white":
        white_cnt += 1
        points += 20
    elif ball == "black":
        black_cnt += 1
        points = math.floor(points / 2)
    else:
        others_cnt += 1

print(f"Total points: {points}")
print(f"Red balls: {red_cnt}")
print(f"Orange balls: {orange_cnt}")
print(f"Yellow balls: {yellow_cnt}")
print(f"White balls: {white_cnt}")
print(f"Other colors picked: {others_cnt}")
print(f"Divides from black balls: {black_cnt}")
